### PR TITLE
Add a stress-test script for testing RecordingTestMetadata

### PR DESCRIPTION
**Pull request by jasonLaster**
### ⚠️ This PR includes lots of changes to our protocol that we probably won't want to merge

Just pushed a relatively simple test harness for RecordingTestMetadata. The goal is to be able to exercise the logic with a comprehensive set of real world recordings.

**Here's how it works**
* fetch recording test metadata
* call processGroupedTestCases with a replay client
* saves the metadata, annotations, and result to a fixture file
* on subsequent calls, reads the metadata and annotations from a file and compares the results

This is similar in many ways to our visual snapshot tests and is your basic contract test aka my favorite type of test.

**How to use it**
1. Get the support@replay.io api key ([docs](https://www.notion.so/replayio/Logging-in-as-support-replay-io-960f606f7d7b4b61872ddb6ce7003ced)) and add it to `.env` `STRESS_TEST_API_KEY=<api-key>`
2. run `ts-node packages/shared/test/stress-test.ts`
  2.a  run `ts-node packages/shared/test/stress-test.ts 1bf12a5d` (just one recoring) 
  2.b  run `ts-node packages/shared/test/stress-test.ts mnite` (just one workspace)

**Additional comments**
* [ ] I'd like to call the TestSuites cache directly, but I was not sure what i'd need to do to get GraphQL to work. The fact that I could pass in the replayClient was really nice.
* [ ] I'd like to find an alternative to writing fixture files because I don't want to have to check in all of these files so that they'd be available in CI.
* [ ] I'd like to find a good list of recordings to test with. My first thought is something like the cypress kitchen sink or realworld app could be good.
* [ ] It'd be nice to switch to Jest where we'd benefit from snapshot diffing, jest-cases, and the existing harness

### CODE REVIEW

**Comment by jasonLaster** [comment]
There are a couple of commits in here where I made changes to RecordingTestMetadata to prevent it from throwing asserts. I don't plan on landing these. They're just here to get more interesting data for the time being to see what we're missing if we filtered out those the steps..

File: packages/protocol/socket.ts [hunk 1]
```diff
@@ -148,6 +148,10 @@ type SessionCallbacks = {
 
 let gSessionCallbacks: SessionCallbacks | undefined;
 
+export function resetSession() {
+  gSessionCallbacks = undefined;
+}
+
```
Comment (jasonLaster) [inline packages/protocol/socket.ts:154]: This was a hack needed to run multiple tests. There's probably a better way

File: packages/shared/client/ReplayClient.ts [hunk 2]
```diff
@@ -244,6 +249,11 @@ export class ReplayClient implements ReplayClientInterface {
     return sessionId;
   }
 
+  releaseSession(): void {
+    client.Recording.releaseSession({ sessionId: this.getSessionIdThrows() });
+    resetSession();
+  }
```
Comment (jasonLaster) [inline packages/shared/client/ReplayClient.ts:255]: needed to test multiple recordings. not sure if this is the best approach or not

File: packages/shared/test-suites/RecordingTestMetadata.ts [hunk 3]
```diff
@@ -411,8 +411,10 @@ export async function processCypressTestRecording(
         }
 
         const annotations = userActionEventIdToAnnotations[id];
-
-        assert(annotations != null, `Missing annotations for test event (${command.name})`);
+        if (!annotations) {
+          // console.warn(`No annotations: ${id}`, command);
+          return;
+        }
```
Comment (jasonLaster) [inline packages/shared/test-suites/RecordingTestMetadata.ts:417]: helpful for seeing what we'd miss if we didn't throw. I dont plan on landing this.

File: packages/shared/test-suites/RecordingTestMetadata.ts [hunk 4]
```diff
@@ -464,41 +466,45 @@ export async function processCypressTestRecording(
...
-          },
-          type: "user-action",
-        });
+            timeStampedPointRange: {
+              begin: beginPoint,
+              end: endPoint,
+            },
+            type: "user-action",
+          });
+        } else {
```
Comment (jasonLaster) [inline packages/shared/test-suites/RecordingTestMetadata.ts:495]: same as above

File: packages/shared/test-suites/RecordingTestMetadata.ts [hunk 5]
```diff
@@ -578,9 +584,10 @@ export async function processGroupedTestCases(
             }
 
             return accumulated;
-          },
-          []
-        );
+          }, [])
+          .filter(annotations =>
+            annotations.find(annotation => annotation.message.event === "test:end")
+          );
```
Comment (jasonLaster) [inline packages/shared/test-suites/RecordingTestMetadata.ts:590]: same as above

File: packages/shared/test/stress-test.ts [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+    );
+
+    updateFixture(recordingId, { ...fixture, result });
+    await client.releaseSession();
+
+    if (fixture.result) {
+      return isEqual(fixture.result, result);
+    }
+
+    return true;
```
Comment (jasonLaster) [inline packages/shared/test/stress-test.ts:141]: it's nice that the test actually fits in a couple lines of code. The rest is just boilerplate

**Comment by bvaughn** [comment]
Cleaning up our Pull Requests today. This is an old draft PR that seems to have been dropped, so I'm going to close it for now. Please re-open if that's a mistake.

**Pull request closed without merging**